src/main/java/com/tw/go/plugin/provider/github/GitHubProvider.java

```diff
@@ -82,4 +77,42 @@ public User getUser(Profile profile) {
         }
         return users;
     }
+
+    @Override
+    public boolean authorize(PluginSettings pluginSettings, User user) {
+        if(StringUtils.isEmpty(pluginSettings.getUsernameRegex())) {
+            return true;
+        } else {
+            return isAMemberOfOrganization(pluginSettings, user);
+        }
+    }
+
+    private boolean isAMemberOfOrganization(PluginSettings pluginSettings, User user) {
+        boolean result = false;
+        try {
+            GitHub github = getGitHub(pluginSettings);
+
+            GHOrganization organization = github.getOrganization(pluginSettings.getUsernameRegex());
```

Comment:
> I think support 1 organization is enough for most of usages and we can improve in the future if somebody ask.

Ok.

> Btw, we should rename the field "username_regex" if you are interest in merge this PR.

yes im planning to have separate plugin settings for github & google, rename username_regex to organizations.